src/Compilers/Core/Portable/SourceGeneration/GeneratorDriver.cs

```diff
@@ -320,10 +325,19 @@ private static GeneratorState SetGeneratorException(CommonMessageProvider provid
 
         internal static string GetFilePathPrefixForGenerator(ISourceGenerator generator)
         {
-            var type = generator.GetType();
+            var type = GetGeneratorType(generator);
             return Path.Combine(type.Assembly.GetName().Name ?? string.Empty, type.FullName!);
         }
 
+        internal static Type GetGeneratorType(ISourceGenerator generator)
```

Comment:
So the other method is public but this isn't -- is this just the easier one to make public?